lib/teiserver/battle/balance/split_one_chevs.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Teiserver.Battle.Balance.SplitOneChevs do
+  alias Teiserver.CacheUser
+  alias Teiserver.Account
+
+  @moduledoc """
+    This balance algorithm first sorts the users by visible OS (match rating) descending. Then all rank=0 (one chevs) will be placed at the bottom of this sorted list.
+
+    Next a team will be chosen to be the picking team. The picking team is the team with the least amount of players. If tied, then the team with the lowest total rating.
+
+    Next the picking team will pick the player at the top of the sorted list.
+
+    This is repeated until all players are chosen.
+
+    This algorithm completely ignores parties.
+
+  """
+
+  @doc """
+  Input:
+  expanded_group:
+  [
+      %{count: 2, members: [1, 4], group_rating: 13, ratings: [8, 5]},
+      %{count: 1, members: [2], group_rating: 6, ratings: [6]},
+      %{count: 1, members: [3], group_rating: 7, ratings: [7]}
+  ]
+  """
+  def perform(expanded_group, team_count, _opts \\ []) do
+    members = flatten_members(expanded_group) |> sort_members()
+    %{teams: teams, logs: logs} = assign_teams(members, team_count)
+    standardise_result(teams, logs)
+  end
+
+  @doc """
+  Input:
+  expanded_group:
+  [
+      %{count: 2, members: [1, 4], group_rating: 13, ratings: [8, 5]},
+      %{count: 1, members: [2], group_rating: 6, ratings: [6]},
+      %{count: 1, members: [3], group_rating: 7, ratings: [7]}
+  ]
+
+  Output:  [
+  %{rating: 8, member_id: 1},
+  %{rating: 5, member_id: 4},
+  %{rating: 6, member_id: 2},
+  %{rating: 7, member_id: 3}
+  ]
+  """
+  def flatten_members(expanded_group) do
+    for %{members: members, ratings: ratings} <- expanded_group,
+        # Zipping will create binary tuples from 2 lists
+        {id, rating} <- Enum.zip(members, ratings),
+        # Create result value
+        rank = get_rank(id),
+        do: %{member_id: id, rating: rating, rank: rank}
+  end
+
+  def get_rank(member_id) do
+    CacheUser.calculate_rank(member_id, "Playtime")
```

Comment:
@FIr3baL this is a mistake in my code. Since you can set ranking method via admin > site config, I need to make this more dynamic. I will change this.

Reply:
@FIr3baL have updated the branch to fix the calculation of rank. If you can pull the latest branch when you have time thanks.